File: back/src/main/java/mitei/mitei/political/balancesheet/manage/kanrensha/dto/FrameworkCapsuleDto.java
package mitei.mitei.political.balancesheet.manage.kanrensha.dto;

import java.io.Serializable;

import mitei.mitei.political.balancesheet.manage.kanrensha.dto.sequrity.UserPersonLeastDto;

/**
 * (Frameworkという名称だが実態は仮)CapsuleDto
 */
public class FrameworkCapsuleDto implements Serializable {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** 最小限ユーザDto */
    private UserPersonLeastDto userPersonLeastDto = new UserPersonLeastDto();

    /**
     * 最小限ユーザDtoを取得する
     *
     * @return 最小限ユーザDto
     */
    public UserPersonLeastDto getUserPersonLeastDto() {
        return userPersonLeastDto;
    }

    /**
     * 最小限ユーザDtoを設定する
     *
     * @param userPersonLeastDto 最小限ユーザDto
     */
    public void setUserPersonLeastDto(final UserPersonLeastDto userPersonLeastDto) {
        this.userPersonLeastDto = userPersonLeastDto;
    }

}

File: front/src/dto/frameworkCapsuleDto.ts
import type UserPersonLeastInterface from "./user/userPersonLeastDto";
import UserPersonLeastDto from "./user/userPersonLeastDto";

export default interface FrameworkCapsuleInterface {

}


export default class FrameworkCapsuleDto implements FrameworkCapsuleInterface {

    /** 最小限ユーザDto */
    userPersonLeastDto: UserPersonLeastInterface;
    
   constructor() {
        this.userPersonLeastDto = new UserPersonLeastDto();
    }
}
